Mage.Sets/src/mage/cards/w/WakeToSlaughter.java

```diff
@@ -0,0 +1,103 @@
+package mage.cards.w;
+
+import mage.MageObject;
+import mage.abilities.Ability;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.keyword.FlashbackAbility;
+import mage.cards.*;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.Zone;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.players.Player;
+import mage.target.Target;
+import mage.target.TargetCard;
+import mage.target.common.TargetOpponent;
+
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * @author LePwnerer
+ */
+public final class WakeToSlaughter extends CardImpl {
+
+    public WakeToSlaughter(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{3}{B}{R}");
+
+        // Choose up to two target creature cards in your graveyard. An opponent chooses one of them. Return that card to your hand. Return the other to the battlefield under your control. It gains haste. Exile it at the beginning of the next end step.
+        this.getSpellAbility().addEffect(new WakeToSlaughterEffect());
```

Comment:
this needs to choose the targets here as part of the spell being cast and not resolution